src/mon/MonCommands.h

```diff
@@ -447,7 +447,8 @@ COMMAND("osd dump " \
 	"print summary of OSD map", "osd", "r", "cli,rest")
 COMMAND("osd tree " \
 	"name=epoch,type=CephInt,range=0,req=false " \
-	"name=states,type=CephChoices,strings=up|down|in|out,n=N,req=false", \
+        "name=states,type=CephChoices,strings=up|down|in|out,n=N,req=false " \
+        "name=shadow,type=CephChoices,strings=--show-shadow,req=false", \
```

Comment:
Actually, thinking about this a bit more, I think we should leave the 'osd tree' commadn alone, and instead add this informaiton to the 'osd crush tree' command (which is also going to show the weight-set values for each pool)